protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {IAtomicBridgeInitiator} from "./IAtomicBridgeInitiator.sol";
+import {IWETH9} from "./IWETH9.sol";
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+
+contract AtomicBridgeInitiator is IAtomicBridgeInitiator, Initializable {
+    struct BridgeTransfer {
+        uint256 amount;
+        address originator;
+        bytes32 recipient;
+        bytes32 hashLock;
+        uint256 timeLock;
+        bool completed;
+    }
+
+    mapping(bytes32 => BridgeTransfer) public bridgeTransfers;
+    bytes32[] public bridgeTransferIds;
```

Comment:
When we opt for this solution, we should definitely avoid storing all  transfers, as this could become excessively large over time. Consequently, since you have chosen this approach, it means we cannot iterate over the mapping, as such my intuition was correct. Therefore, let's redesign the requirements here.

Reply:
But we need to store all this data on chain, so that indexers could show historical data. The cost of storing data in a mapping does not directly depend on the "height" or the number of entries already present in the mapping. The cost of storing a new entry in a mapping is constant, meaning it does not increase as more entries are added

 Many contracts store all their data in a mapping.

Reply:
A mapping in solidity is a hash table